2022/Q2/artifacts/pos-model/PoS-model.md

```diff
@@ -294,15 +276,13 @@ func new_evidence(evidence){
 ```
 
 ```go
-//COMMENT: shall we do updates to this state to happen once at the end of an epoch? This has been discussed.
-//Tomas agreed is interesting, but they are not doing it rigth know.
-//COMMENT: still unclear from the docs. Why are unbonds substracted? When creating unbonds, we already decrement bonds,
-//so we should not substract them here again, unless I am missing something. This is now resolved: there was a typo in the spec.
+/* COMMENT
+shall we do updates to this state to happen once at the end of an epoch? This has been discussed.
+Tomas agreed is interesting, but they are not doing it rigth know.
```

Comment:
You may want to add to this comment that doing it at the end may save gas for repetitive transactions in the same epoch

Reply:
I am not sure about it. The problem of doing it at then end is that one has to iterate over all validators. While this will be amortized if most of the validators execute PoS transaction in a given epoch, it will be expensive if in practice only a few do. I was actually thinking of removing the comment completely, given that Chris said it was important for them to avoid those iterations. What do you think?

Reply:
Hmmm. I think I agree. Hadn't thought about it from that perspective